internal/internal.go

```diff
@@ -62,6 +62,8 @@ var (
 	// gRPC server. An xDS-enabled server needs to know what type of credentials
 	// is configured on the underlying gRPC server. This is set by server.go.
 	GetServerCredentials any // func (*grpc.Server) credentials.TransportCredentials
+	// GetConnection gets the connection from the context.
+	GetConnection any // func (context.Context) net.Conn
```

Comment:
It should be fine to give this a type to avoid the casts, since it's only things from the stdlib.

But, it should also be fine to implement it fully within the internal package.  But then.. why not just leave it where it is and export the setter?  Or find another place in internal for it?

Reply:
Thought about this, decided to move back to transport. I don't really think any of the current internal directories make semantical sense, and as per offline discussion I think there's complications about trying to keep file paths consistent.